### DISCUSSION TITLE
How does the TTL worked

### DISCUSSION THREAD

**Discussion by bingbb47** [milvus-io/milvus#26502] [Q&A and General discussion] [answered]
If I set the TTL property to a collection that already have some items, how does it work?
It will drop the “out of date”  items immediately and keep the live items , or it will drop all of the items after a ttl?

I have try this by some tests, but it seems not effective at the time, at least It was not reflected in the `collection.num_entities` and    `collection.query`. The items seems only be dropprd after compact or flush.

so, could anyone can say something more about ttl?

**Comment by yhmo** [comment]
In Milvus, delete action is "soft-delete". The deleted items are marked as "soft-delete" and cannot be searchable. Only after the segment is compacted, the items are deleted from memory/disk permanently.


And, collection.num_entities doesn't count the deleted items. This is a technical limitation. In v2.3, we will provide a new method to calculate accurate row count.

**Reply by bingbb47** [reply, marked as answer]
So, the expiration operation is equivalent in behavior to marking an item as deleted. When i set the ttl in a moment, the items that meet the expiration time will be marked as "delete" and wait for compact.
